src/v3/src/components/CaptchaContainer/CaptchaContainer.tsx

```diff
@@ -0,0 +1,135 @@
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import HCaptcha from '@hcaptcha/react-hcaptcha';
+import { Box } from '@okta/odyssey-react-mui';
+import { useEffect, useRef } from 'preact/hooks';
+import ReCAPTCHA from 'react-google-recaptcha';
+
+import { useWidgetContext } from '../../contexts';
+import { useOnSubmit } from '../../hooks';
+import {
+  CaptchaContainerElement,
+  UISchemaElementComponent,
+} from '../../types';
+
+const CaptchaContainer: UISchemaElementComponent<{
+  uischema: CaptchaContainerElement
+}> = ({ uischema }) => {
+  const {
+    options: {
+      siteKey,
+      type: captchaType,
+      captchaId,
+    },
+  } = uischema;
+
+  const { dataSchemaRef } = useWidgetContext();
+  const onSubmitHandler = useOnSubmit();
+  const dataSchema = dataSchemaRef.current!;
+  const captchaRef = useRef<ReCAPTCHA | HCaptcha>(null);
+
+  useEffect(() => {
+    // set the reference in dataSchema context to this captcha instance
+    dataSchema.captchaRef = captchaRef;
+    //  For some reason Hcaptcha does not initially auto render only when in development mode
+    //  This delayed manual render is only used in development mode for hcaptcha
+    if (process.env.NODE_ENV === 'development' && captchaType === 'HCAPTCHA') {
+      setTimeout(() => {
+        (captchaRef.current as HCaptcha)?.renderCaptcha();
+      }, 500);
+    }
+    return () => {
+      dataSchema.captchaRef = undefined;
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dataSchema]);
+
+  const onErrorCaptcha = (e: Event | string) => {
+    console.error(e);
+  };
+
+  const resetCaptchaContainer = () => {
+    if (captchaType === 'RECAPTCHA_V2') {
+      (captchaRef.current as ReCAPTCHA)?.reset();
+    }
+    (captchaRef.current as HCaptcha)?.resetCaptcha();
+  };
+
+  const onVerifyCaptcha = (token: string | null) => {
+    if (!token) {
+      console.error('captcha token expired');
+      return;
+    }
+
+    const {
+      submit: {
+        actionParams: params,
+        step,
+        includeImmutableData,
+      },
+    } = dataSchema;
+
+    // insert received captcha success token into the captchaVerify submit param
+    const captchaSubmitParams = {
+      captchaVerify: {
+        captchaToken: token,
+        captchaId,
+      },
+      ...params,
+    };
+
+    // Captcha challenge solved successfully, submit form
+    onSubmitHandler({
+      includeData: true,
+      includeImmutableData,
+      params: captchaSubmitParams,
+      step,
+    });
+
+    // tokens are one-time use, so if we submit the same token twice then it will be rejected
+    // by the server the second time, so we must reset the captcha state
+    resetCaptchaContainer();
+  };
+
+  if (captchaType === 'RECAPTCHA_V2') {
+    return (
+      // set z-index to 9999 for ReCaptcha so the badge does not get covered by the footer
+      <Box
+        zIndex={9999}
```

Comment:
This is a bit overkill, we should find the proper z-index that is high enough to avoid being covered by the footer rather than picking a really big number

Reply:
Actually, what is the "badge" that was getting covered by the footer? Can you provide a screenshot of this scenario?

Reply:
<img width="1204" alt="image" src="https://github.com/okta/okta-signin-widget/assets/107433508/cb06607d-5701-47f2-bbc2-5b6460c62b05">

Its the little badge in the bottom right corner that expands out a little bit when you hover over it.  I found that a `z-index` of `1` is all that is needed to show on top of the footer.  The footer must have a z-index of 0.